websites/views.py

```diff
@@ -25,6 +26,7 @@ class DefaultPagination(LimitOffsetPagination):
 
 class WebsiteViewSet(
     mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
```

Comment:
I think we might need 2 separate viewsets for the Website model.  One read-only viewset, accessible to unauthenticated users, currently used for listing newly published websites on the front page.  And another requiring authentication, to list and view/edit details only for the websites the user has view/edit access to (`WebsiteEditorViewSet`), that does not exclude unpublished websites?  For example, right now if I try to bring up the detail API view for an unpublished website I get a 404 response - the queryset excludes them:

```python
        queryset = Website.objects.filter(
            publish_date__lte=now_in_utc(),
        ).order_by(ordering)
```

Reply:
We shouldn't need a separate viewset. We should be able to manipulate the queryset based on the request user

Reply:
@mbertrand Should we filter unpublished websites in this PR? I can make the change now if so

Reply:
Either way is fine with me (I guess it would be changed to filtering by published only for anonymous users?). If you prefer I can do that as part of the work on integrating the permission classes with the views, once this PR and the permissions PR are merged.

Reply:
Sure, I'll leave it to you since it might be easier to deal with the views all at once

Reply:
Okay, it looks like the API is already broken anyway on CI - https://github.com/mitodl/ocw-studio/issues/59